Docs/RHCL-RELEASE-GUIDE.md

```diff
@@ -0,0 +1,873 @@
+     - [`fbc-standard`](https://gitlab.cee.redhat.com/releng/konflux-release-data/-/blob/main/config/common/product/EnterpriseContractPolicy/fbc-standard.yaml) (file-based catalog releases)
+
+6. **Constraints**
+   - Located in: [`releng/konflux-release-data/constraints/product/api-management.yaml`](https://gitlab.cee.redhat.com/releng/konflux-release-data/-/blob/main/constraints/product/api-management.yaml)
+   - Validates RPA values (registries, tags, policies)
+
+7. **Pyxis Delivery Repositories**
+   - All target repositories must be defined in Pyxis before release
+   - Managed via: `releng/pyxis-repo-configs` repository
+
+---
+
+## Step-by-Step: Releasing RHCL Components
+
+This section covers releasing operator images and bundles (not FBC).
+
+### Step 1: Verify Builds are Complete
+
+1. **Check PipelineRuns**
+   ```bash
+   oc project api-management-tenant
+   oc get pipelineruns --sort-by=.metadata.creationTimestamp | tail -20
+   ```
+
+2. **Find Recent Successful Builds**
+   - Look for PipelineRuns with `Succeeded` status
+   - Note the Snapshot name (format: `{application}-{hash}`)
+
+3. **Inspect Snapshot**
+   ```bash
+   oc get snapshot {snapshot-name} -o yaml
+   ```
+   - Verify all expected components are present
+   - Note the component image digests
+
+### Step 2: Choose Staging or Production
+
+- **Staging Release**: Test the release process without affecting customers
+  - Target: `registry.stage.redhat.io`
+  - RPA files end with: `-stage-rhel9.yaml`
+  - Example: `rhcl-1-2-rhcl-operator-stage-rhel9.yaml`
+
+- **Production Release**: Customer-facing release
+  - Target: `registry.redhat.io`
+  - RPA files end with: `-prod-rhel9.yaml`
+  - Example: `rhcl-1-2-rhcl-operator-prod-rhel9.yaml`
+
+### Step 3: Update or Create ReleasePlanAdmission
+
+**Important**: You will need to create a new RPA for each release of new components.
+
+For existing components with updated versions:
+
+1. **Clone the Release Data Repo**
+   ```bash
+   git clone git@gitlab.cee.redhat.com:releng/konflux-release-data.git
+   cd konflux-release-data
+   git checkout -b update-rhcl-1.2.0
+   ```
+
+2. **Edit the RPA File**
+   ```bash
+   vim config/stone-prd-rh01.pg1f.p1/product/ReleasePlanAdmission/api-management/rhcl-1-2-rhcl-operator-prod-rhel9.yaml
+   ```
+
+3. **Update Component Versions**
+   - Locate the `mapping.components` section
+   - Update `tags` for each component
+   - Example:
+     ```yaml
+     - name: rhcl-1-2-rhcl-operator
+       repository: "registry.redhat.io/rhcl-1/rhcl-rhel9-operator"
+       tags:
+         - "v0.2.1-{{digest_sha}}"  # New version
+         - "0.2.1"
+         - "rhcl-1.2.0"  # New product version tag
+     ```
+
+4. **Update Release Notes**
+   ```yaml
+   data:
+     releaseNotes:
+       product_id: [979]
```

Comment:
These fields will never change unless we change major version, so this section can probably be deleted